packages/build-utils/src/fs/run-user-scripts.ts

```diff
@@ -299,9 +299,13 @@ export async function scanParentDirs(
   let lockfileVersion: number | undefined;
   let cliType: CliType = 'yarn';
 
-  const [hasYarnLock, packageLockJson, pnpmLockYaml, bunLockBin] =
+  const hasYarnLock = Boolean(yarnLockPath);
+
+  const [yarnLockYaml, packageLockJson, pnpmLockYaml, bunLockBin] =
     await Promise.all([
-      Boolean(yarnLockPath),
+      yarnLockPath
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (non-blocking):** ‏definitely not a big deal, but we do already have a slighty more explicit variable

```suggestion
      hasYarnLock
```

Reply:
it's not the same thing. `hasYarnLock` was being used to flag through the existence of the lockfile. Now it's the actual lockfilePath, (similar to the other paths). I moved the `hasYarnLockfile` binding further down